okapi-core/src/main/java/org/folio/okapi/managers/ProxyService.java

```diff
@@ -203,9 +203,20 @@ private boolean resolveRedirects(ProxyContext pc,
    */
   private boolean checkTokenCache(String tenant, HttpServerRequest req, String pathPattern,
       ModuleInstance mi) {
-    CacheEntry cached =
-        tokenCache.get(tenant, req.method().name(), pathPattern == null ? req.path() : pathPattern,
-            req.getHeader(XOkapiHeaders.USER_ID), req.headers().get(XOkapiHeaders.TOKEN));
+    String token = req.headers().get(XOkapiHeaders.TOKEN);
+    if (token == null) {
+      mi.setAuthToken(null);
+      mi.setUserId(null);
+      mi.setPermissions(null);
+      return false;
+    }
+
+    String pathComponent = pathPattern == null ? req.path() : pathPattern;
+
+    CacheEntry cached = tokenCache == null ? null
+        : tokenCache.get(tenant, req.method().name(),
```

Comment:
Checking for tokenCache == null seems redundant. It is always constructed. Perhaps it should be final?